src/sobject/sobject.cpp

```diff
@@ -0,0 +1,171 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#include <boost/algorithm/string.hpp>
+#include <mutex>
+#include "sisl/logging/logging.h"
+#include "sisl/sobject/sobject.hpp"
+
+namespace sisl {
+
+sobject_ptr sobject::get_child(const std::string& name) {
+    std::shared_lock lock{m_mtx};
+    for (const auto& [id, obj] : m_children) {
+        if (id.name == name) { return obj; }
```

Comment:
Can you also comment here why we return the first match would help for readers.

Reply:
Ack